src/Mono.Android/Android.Runtime/AndroidRuntime.cs

```diff
@@ -278,30 +278,32 @@ protected override IEnumerable<Type> GetTypesForSimpleReference (string jniSimpl
 
 		protected override string? GetSimpleReference (Type type)
 		{
+			if (RuntimeFeature.UseReflectionForManagedToJava) {
+				return JavaNativeTypeManager.ToJniName (type);
+			}
```

Comment:
I could actually move this check _inside_ `TypemapManagedToJava()`, is that actually better?

Reply:
I'll try this to see if that is even simpler.